platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -367,18 +373,27 @@ OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
 {
     OtaPalStatus_t result = OTA_PAL_COMBINE_ERR( OtaPalUninitialized, 0 );
     char realFilePath[ OTA_FILE_PATH_LENGTH_MAX ];
+    char * pFileName = NULL;
 
     if( C != NULL )
     {
         if( C->pFilePath != NULL )
         {
             if( C->pFilePath[ 0 ] != ( uint8_t ) '/' )
             {
-                /* POSIX port using standard library */
-                /* coverity[misra_c_2012_rule_21_6_violation] */
-                int res = snprintf( realFilePath, OTA_FILE_PATH_LENGTH_MAX, "%s/%s", getenv( "PWD" ), C->pFilePath );
-                assert( res >= 0 );
-                ( void ) res; /* Suppress the unused variable warning when assert is off. */
+                /* Get current directory. */
+                pFileName = getcwd( realFilePath, OTA_FILE_PATH_LENGTH_MAX );
+
+                if( pFileName == NULL )
+                {
+                    LogError( ( "Failed to get current working directory: %s", strerror( errno ) ) );
+                }
+                else
+                {
+                    /* Add the filename . */
+                    realFilePath[ strlen( realFilePath ) ] = '/';
+                    strncat( realFilePath, ( char * ) C->pFilePath, strlen( ( const char * ) C->pFilePath ) + 1U );
```

Comment:
This call to `strncat()` will produce unpredictable results since `realFilePath` is no longer NULL-terminated after line 394.  In addition, you haven't verified that the storage for `realFilePath` is sufficient to hold the fully-qualified pathname you're building here.

To fix this, I would:

1. Change line 385 to: `pFileName = getcwd( realFilePath, OTA_FILE_PATH_LENGTH_MAX - 1 );`
1. Change lines 394-395 as follows:

```c
strncat( realFilePath, "/", 1 );  /* realFilePath is still NULL-terminated */
if( strlen( realFilePath ) + strlen( C->pFilePath ) < OTA_FILE_PATH_LENGTH_MAX ) )
{
    strncat( realFilePath, C->pFilePath, strlen( C->pFilePath ) );
}
```

Also note that you don't want to add 1 to the size (3rd argument) passed to `strncat()` here.  `man strncat` to see why.

Reply:
Thanks, extracted file path generation into `getFilePathFromCWD()` and checked if the buffer is large enough to hold filepath + filename + `\` before performing `strncat()`